tally_ho/apps/tally/static/js/barcode_verify.js

```diff
@@ -32,41 +65,101 @@ function hide_barcode_show_placeholder() {
-    $("#id_barcode_copy").focusout(function(evt) {
+    $("#id_barcode_copy").focusout(function (evt) {
         if ($("#id_barcode_copy").val().length == $("#id_barcode").val().length &&
-                $('#id_barcode').val() != $("#id_barcode_copy").val()) {
+            $('#id_barcode').val() != $("#id_barcode_copy").val()) {
             show_barcode_hide_placeholder();
         }
     });
 
-    $("#id_barcode_copy").keyup(function(evt) {
+    $("#id_barcode_copy").keyup(function (evt) {
         if ($("#id_barcode_copy").val().length == $("#id_barcode").val().length &&
-                $("#id_barcode").val() == $("#id_barcode_copy").val()) {
+            $("#id_barcode").val() == $("#id_barcode_copy").val()) {
             show_barcode_hide_placeholder();
             $("#id_barcode").parent().addClass('has-success');
             $("#id_barcode_copy").parent().addClass('has-success');
         } else if ($("#id_barcode_copy").val().length == $("#id_barcode").val().length) {
             show_barcode_hide_placeholder();
+            $("#id_barcode").parent().removeClass('has-success');
+            $("#id_barcode_copy").parent().removeClass('has-success');
             $("#id_barcode").parent().addClass('has-error');
             $("#id_barcode_copy").parent().addClass('has-error');
         } else {
             hide_barcode_show_placeholder();
         }
     });
 
-    $("#id_barcode_copy").focusin(function(evt) {
+    $("#id_barcode_copy").focusin(function (evt) {
         hide_barcode_show_placeholder();
     });
 });
+
+function add_element(elementTag, elementId, html) {
+    const table = document.getElementById("formtablebody");
+    const newElement = document.createElement(elementTag);
+    newElement.setAttribute('id', elementId)
+    newElement.innerHTML = html;
+    table.insertBefore(newElement, table.firstChild);
+}
+
+
+function change_barcode_entry_mode(barcode_entry_mode) {
+    const result_form = document.getElementById('result_form');
+    const id_form_instructions = document.getElementById('id_form_instructions');
+    const barcode_manual_entry = document.getElementById('barcode_manual_entry');
+    const barcode_scanned_entry = document.getElementById('barcode_scanned_entry');
+    const barcode_scan_input_field = () => document.getElementById('id_scanned_barcode');
+    const barcode_manual_input_field = () => document.getElementById('id_barcode');
+    const manual_entry_button = () => document.getElementById('manual_entry_button');
+    const scanned_entry_button = () => document.getElementById('scanned_entry_button');
+    const barcode_copy_manual_entry = document.getElementById('barcode_copy_manual_entry');
+    const table = document.getElementById("formtablebody");
+    const barcode_errors_section = () => document.getElementById('barcode_errors');
+
+    if (barcode_entry_mode == 'manual') {
+        table.removeChild(barcode_scanned_entry)
+        barcode_input_html = '<td><label>Barcode</label></td>' +
```

Comment:
instead of editing the DOM here, can we have this always rendered but be hidden by default?

Reply:
yes this makes sense since it never changes.

Reply:
on a second thought this won't be possible since it's inside a `tr tag ` which are being deleted and recreated when switching between the entry modes.

Reply:
Can we then have 2 TR tags, each with a different ID and toggle which one is visible?